graphics/deko3d/deko_examples/source/SampleFramework/CIntrusiveTree.h

```diff
@@ -194,31 +188,11 @@ class CIntrusiveTree final : protected CIntrusiveTreeBase
             mode != UpperBound ? N::Left : N::Right,
             [&lambda](N* curnode) { return lambda(toType(curnode)); });
 
-        switch (mode)
-        {
-            default:
-            case Exact:
-                break;
-            case LowerBound:
-                if (!node && parent)
-                {
-                    if (&parent->left() == &point)
-                        node = parent;
-                    else
-                        node = walk(parent, N::Right);
-                }
-                break;
-            case UpperBound:
-                if (node)
-                    node = walk(node, N::Right);
-                else if (parent)
-                {
-                    if (&parent->right() == &point)
-                        node = walk(parent, N::Right);
-                    else
-                        node = parent;
-                }
-                break;
+        if (mode != Exact) {
+            if (mode == UpperBound && node)
+                node = walk(node, N::Right);
+            else if (!node && parent)
+                node = &parent->left() == &point ? parent : walk(parent, N::Right);
```

Comment:
Is this ternary actually correct for UpperBound mode?

Reply:
Please let me know if there is any problem.

Reply:
I was mostly asking how did you arrive to that simplification, as I'm currently struggling to wrap my head around this code :p

Reply:
After some analysis, I believe this code to be correct.